mypy/constraints.py

```diff
@@ -1018,21 +1018,25 @@ def visit_callable_type(self, template: CallableType) -> list[Constraint]:
             param_spec = template.param_spec()
 
             template_ret_type, cactual_ret_type = template.ret_type, cactual.ret_type
+            bool_type = UnionType(
+                [LiteralType(True, cactual_ret_type), LiteralType(False, cactual_ret_type)]  # type: ignore[arg-type]
+            )
+
             if template.type_guard is not None and cactual.type_guard is not None:
                 template_ret_type = template.type_guard
                 cactual_ret_type = cactual.type_guard
             elif template.type_guard is not None:
                 template_ret_type = AnyType(TypeOfAny.special_form)
```

Comment:
Does this branch make sense?
When we use `TypeGuard` or `TypeIs` in `template` it seems that we should use `TypeGuard`/`TypeIs` in `actual` too. Otherwise, we get `Never` type in the previous steps. Is it true? I'm a little confused here...

Reply:
Yes, it's probably better to remove this branch too. I think when I implemented TypeIs I assumed we needed this to make `bool` and `TypeGuard`/`TypeIs` unify, but that doesn't seem to be the case.

Reply:
Got it, thanks for your comments! Removed that `elif`-s for `TypeGuard`/`TypeIs`.